ADBench/run-all.ps1

```diff
@@ -155,232 +155,211 @@ $datadir = "$dir/data"
-		#>
-
-		$this.name = $name
-		$this.type = $type
-		$this.objectives = $objectives
-		$this.gmm_both = $gmm_both
-		$this.gmm_use_defs = $gmm_use_defs
-		$this.eigen_config = $eigen_config.ToCharArray() | % { $_ -band "1" }
-	}
-
-	# Run all tests for this tool
-	[void] runall () {
-		Write-Host $this.name
-		if ($this.objectives.HasFlag([ToolType]::GMM)) { $this.testgmm() }
-		if ($this.objectives.HasFlag([ToolType]::BA)) { $this.testba() }
-		if ($this.objectives.HasFlag([ToolType]::Hand)) { $this.testhand() }
-		if ($this.objectives.HasFlag([ToolType]::LSTM)) { $this.testlstm() }
-	}
-
-	# Run a single test
-	[void] run ([string]$objective, [string]$dir_in, [string]$dir_out, [string]$fn) {
-		if ($objective.contains("Eigen")) { $out_name = "$($this.name)_Eigen" }
-		elseif ($objective.contains("Light")) { $out_name = "$($this.name)_Light" }
-		elseif ($objective.endswith("SPLIT")) { $out_name = "$($this.name)_split" }
-		else { $out_name = $this.name }
-		$output_file = "${dir_out}${fn}_times_${out_name}.txt"
-		if (!$script:repeat -and (Test-Path $output_file)) {
-			if ($script:repeat_failures) {
-				$test_failed = Select-String -quiet '^inf inf$' $output_file
-				if (!$test_failed) {
-					Write-Host "          Skipped test (already completed, and wasn't a fail)"
-					return
-				}
-			} else {
-				Write-Host "          Skipped test (already completed)"
-				return
-			}
-		}
-
-		$cmd = ""
-		$cmdargs = @($dir_in, $dir_out, $fn, $script:nruns_f, $script:nruns_J, $script:time_limit)
-		if ($this.type -eq "bin") {
-			$cmd = "$script:bindir/tools/$($this.name)/Tools-$($this.name)-$objective.exe"
-		} elseif ($this.type -eq "cpp") {
-            #temporary solution until this script refactoring
-			$cmd = "$script:bindir/src/cpp/runner/CppRunner.exe"
+    [string]$name
+    [ToolType]$type
+    [ObjectiveType]$objectives
+    [bool]$gmm_both
+    [bool]$gmm_use_defs
+
+    # Static constants
+    static [string]$gmm_dir_in = "$datadir/gmm/"
+    static [string]$ba_dir_in = "$datadir/ba/"
+    static [string]$hand_dir_in = "$datadir/hand/"
+    static [string]$lstm_dir_in = "$datadir/lstm/"
+    static [array]$gmm_sizes = @("1k", "10k") # also "2.5M"
+    static [array]$hand_sizes = @("small", "big")
+    static [int]$ba_min_n = 1
+    static [int]$ba_max_n = 5
+    static [int]$hand_min_n = 1
+    static [int]$hand_max_n = 5
+    static [array]$lstm_l_vals = @(2, 4)
+    static [array]$lstm_c_vals = @(1024, 4096)
+    # TODO probably want to set these in CMake somewhere
+
+    # Constructor
+    Tool ([string]$name, [ToolType]$type, [ObjectiveType]$objectives, [bool]$gmm_both, [bool]$gmm_use_defs) {
+        <#
+        .SYNOPSIS
+        Create a new Tool object to be run
+
+        .EXAMPLE
+        [Tool]::new("Finite", "bin", [ObjectiveType] "GMM, BA, Hand, LSTM", 0, 0)
+        This will create a Tool:
+        - called "Finite"
+        - run from binary executables
+        - runs all five tests 
```

Comment:
Does this comment need fixing?  "all five"?

Reply:
Thank you